### ISSUE TITLE
Suggest adding configuration for aria-live on calcite-alert

### ISSUE THREAD

**Issue by FelFly** [Esri/calcite-components#3572]
### Summary

While the rendered calcite-alert component does have a `role="alertdialog"` on it, typically I'd expect to use that in conjunction with an [aria-live region](https://www.w3.org/TR/wai-aria-1.1/#aria-live), especially on auto-dismissing alerts. It appears that the component can support this today through code modification, but I didn't see option for configuration on https://developers.arcgis.com/calcite-design-system/components/alert/. At the very least, I might recommend adding notes to the Accessibility section on how this can be manipulated so others don't have to guess.

### Actual Behavior
Out-of-the-box aria-live isn't set as an attribute at all, so the only way to hear the alert in VoiceOver is to navigate to the appropriate place in the DOM (or for the user-agent to move keyboard focus.)

### Expected Behavior
When `role="alert"` or `role="alertdialog"` is present with an aria-live region, it doesn't matter where keyboard focus is in the page, the screen-reader (in my case, VoiceOver) will read out the message either after their current element has been read (eg. "polite") or as an immediate interruption (eg. "assertive").

### Reproduction Steps and Sample

_Sample_: https://codepen.io/FelFly/pen/VwzJXpq

1. Open https://developers.arcgis.com/calcite-design-system/components/alert/ sample codepen
2. Add a line of paragraph text before the calcite-alert
2. Turn on VoiceOver (CMD+F5)
3. Click onto preview pane
4. Ctrl+Option+right-arrow to move to the paragraph text (you should hear it read)
5. Do not move to the next DOM element using keyboard
6. Observed: You will not hear the alert.
7. Open my codepen above, repeat steps 3-5. You should hear the alert read without having to move focus to it. With `assertive` you should hear it right after or in the middle of whatever paragraph text you wrote. With `polite` you may hear additional navigation instructions in between the paragraph text and alert being read.

### Relevant Info

Safari, VoiceOver

_Version_: `@esri/calcite-components@v1.0.0-beta.69`

- [ ] CDN
- [ ] NPM package

**Comment by macandcheese** [comment]
We might also want to address focus trapping in alerts in relation to this.

**Comment by anveshmekala** [comment]
@macandcheese & @asangma  can we default the `aria-live` value to be `polite` and when `autoDimiss` is enabled set it to `assertive` . Any thoughts?

**Comment by macandcheese** [comment]
I found this example where its configurable via prop in a similar component: https://bootstrap-vue.org/docs/components/toast#comp-ref-b-toast-props

**Comment by anveshmekala** [comment]
> I found this example where its configurable via prop in a similar component: https://bootstrap-vue.org/docs/components/toast#comp-ref-b-toast-props

Was thinking to use the existing user configurable prop `autoDimiss`  to set assertive for `aria-live` as the example did using `role` instead of having a new prop for just the aria-live .

**Comment by jcfranco** [comment]
@FelFly Would the proposal above address the main issue of the alert not being announced when it is auto-dismissable or would you still need fine-grained control over `aria-live`?

**Comment by FelFly** [comment]
@jcfranco So it would default to polite, but use assertive when auto-dismissible? I think that's fine.

**Comment by jcfranco** [comment]
Awesome. Thanks for confirming!

**Comment by anveshmekala** [comment]
@FelFly Is the issue related to safari browser precisely as mentioned ?

**Comment by FelFly** [comment]
@anveshmekala No, not necessarily. I just test with Safari because it's the safest browser combo for VoiceOver, since both are produced by Apple. Essentially I would expect it to "at least" work with Safari.

**Comment by anveshmekala** [comment]
For safari, @jcfranco  couldn't reproduce the issue. As mentioned [here](https://www.w3.org/TR/wai-aria-1.2/#:~:text=Elements%20with%20the%20role%20alert%20have%20an%20implicit%20aria%2Dlive%20value%20of%20assertive%2C%20and%20an%20implicit%20aria%2Datomic%20value%20of%20true.), `role` attribute will set the `aria-live` attribute implicitly. We do see the issue for chrome though. Could you please re-verify ? 
cc @jcfranco

**Comment by jcfranco** [comment]
For context, I tested locally using a simple page using the code from the description in Safari and VO does behave as expected. Could the `<iframe>` be affecting VO's behavior?

**Comment by FelFly** [comment]
Sorry, I missed the notification. Where would you like me to re-verify?

**Comment by jcfranco** [comment]
I meant to share this earlier, but this is what I see on Mac 11.6.1, Safari + VO when I test locally using the markup from the description's codepen (notice how the alert is read by VO as soon as the page loads).

![alert-safari-vo](https://user-images.githubusercontent.com/197440/153521160-1f170b8f-471c-47c6-a995-6c12904811ac.gif)

**Comment by anveshmekala** [comment]
> I meant to share this earlier, but this is what I see on Mac 11.6.1, Safari + VO when I test locally using the markup from the description's codepen (notice how the alert is read by VO as soon as the page loads).

re-verify the issue in safari using VO and if it exists please share a gif or video.

**Comment by jcfranco** [comment]
It looks like this is working as expected. If the browser has focus and you load [the following codepen](https://codepen.io/jcfranco/pen/LYOzLyZ?editors=1010) (based on the repro case), you'll notice that VO reads out the alert as soon as it pops up.

![safari-vo-alert-working-as-expected](https://user-images.githubusercontent.com/197440/153980691-cd25704a-dc52-447e-8f5b-2065e2ca8bb6.gif)

I'm going to go ahead and close this. Feel free to reopen if further action is needed.
